jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/python/PythonLambdaHandlerResolver.kt

```diff
@@ -79,22 +73,11 @@ class PythonLambdaHandlerResolver : LambdaHandlerResolver {
         }
 
         val rootVirtualFile = directory.virtualFile
-        module?.let {
-            val rootManager = ModuleRootManager.getInstance(module)
-            if (rootManager.contentRoots.contains(rootVirtualFile)) {
-                return true
-            }
-
-            if (rootManager.getSourceRoots(false).contains(rootVirtualFile)) {
-                return true
-            }
-
-            if (rootVirtualFile.findChild("requirements.txt") != null) {
-                return true
-            }
+        if (rootVirtualFile.findChild("requirements.txt") != null) {
+            return true
+        } else {
+            throw IllegalStateException("Failed to locate requirements.txt")
```

Comment:
does SAM CLI require a `requirements.txt`? are there any use cases where we did not require this?

Reply:
Technically, no.

```
Building codeuri: /private/tmp/foo/sam-app/hello_world runtime: python3.7 metadata: {} functions: ['HelloWorldFunction']
requirements.txt file not found. Continuing the build without dependencies.
```

 Maybe we revert this and tweak the logic to make requirements.txt higher prio? But today PythonBuilder that reverses the String back to the base path has to have a requirements.txt....so left hand isn't talking to right hand already today...

Let's sync on this offline

Reply:
okay for now let's move forward with this so at least it's consistent and we can re-evaluate how this works later